emission/analysis/userinput/matcher.py

```diff
@@ -97,3 +98,11 @@ def get_user_input_dict(ts, tct, input_key_list):
     logging.debug("for trip %s, returning user input dict %s" % (tct.get_id(), tct_userinput))
     return tct_userinput
 
+# For a given trip, find the corresponding list of label inferences if it has been generated
+def get_inferred_labels_list(trip):
+    candidates = esdt.get_sections_for_trip(esda.INFERRED_LABELS_KEY, trip.user_id, trip.get_id())
+    if len(candidates) == 0: return {}  # Perhaps we have not run the inference step for this trip
+    assert len(candidates) == 1, \
+        "Multiple label inference list objects for trip "+str(trip.get_id())
+    return candidates[0]["data"]["prediction"]
```

Comment:
Let's think through this some more. If we haven't yet run the inference step for this trip, and we run it later, how will those labels assigned to the `confirmed_trip` object? The `last_processed_ts` for the `eaum.create_confirmed_objects` step would have gone past this trip, so we won't consider it again.

For the actual user inputs, we deal with this by having an additional `eaum.match_incoming_user_inputs` whose `last_processed_ts` is keyed to the `write_ts` of the user input objects. (see background at https://github.com/e-mission/e-mission-docs/issues/476#issuecomment-561727669) I don't see similar code here.

Reply:
Cleaned up in https://github.com/e-mission/e-mission-server/pull/825/commits/dee24619a30582baa8c8699fe7ec7ac7edb9bfde.